### ISSUE TITLE
remove_elements error on a coin created inside a purse

### ISSUE THREAD

**Issue by Ptroulitaki2023** [bluefoxr/COINr#61]
Hello Will,

I have been using lately the purse data type in order to conduct a multi-year analysis of the Index I am developing. So I create a purse
```{r}
purse <- new_coin(iData = iData_df,
                  iMeta = iMeta_df,
                  split_to = "all",
                  quietly = TRUE)
```  
and then I perform several steps such as treatment, imputation, normalisation and aggregation, using however several custom techniques. 

When I want to apply remove_elements on the purse but separately to each coin I get the following error

```{r}
ind_impact<-COINr::remove_elements(purse$coin$`2024`, Level=1, dset='Aggregated', iCode="Index")
```
![image](https://github.com/user-attachments/assets/5ab8953e-3428-4dfa-8f92-e9abeaad0052)

I have checked the coins and they do have the _Imputed_ datasets included, so I cannot understand the issue. Moreover, when I develop the coins separately, meaning that I have developed the 2024 Index using the coin datatype only, the `remove_elements()` works properly. I tried to check the differences between `purse$coin$2024` and `CS_2024` and I cannot see any difference in the structure or in the logs.

Do you have any idea on what it might be the issue here?

Thank you in advance,
Petrina

**Comment by bluefoxr** [comment]
Hi Petrina, the issue here is that not all functions support a purse input. The `remove_elements()` function is one of these - you would need to apply it individually to individual coins. Purses are not fully supported everywhere because (a) I didn't have time to integrate them fully everywhere, and (b) in some cases it is not obvious how to extend to a time series. E.g. removing elements calculates the average rank change, but this would have to be averaged over time, or else we have a time-dependent output - there are different ways one could consider doing it.

Probably the best way to approach this is to `lapply()` or similar the function to the coins, then process the list output in whichever way makes most sense to you.

**Comment by Ptroulitaki2023** [comment]
Hello Will,

Thank you very much for your response! It is completely understandable that, given time limitations, it wasn’t possible to fully integrate everything. The package is already thorough and your work much appreciated :).

So, if I understand correctly, even though I am passing to `remove_elements()` a coin object (`purse$coin$2024`), the fact that it was created using purse, is recognised by the function and that is why it throws this error?

I will check your suggestion for a workaround.

Thank you in advance,
Petrina

**Comment by bluefoxr** [comment]
First check that whatever you are passing to the function is definitely a coin class using `class()`. If it still doesn't work, it's probably because what `remove_elements()` is doing behind the scenes is going back to the input data (which is stored in the coin), removing one or more indicators, then re-running all the commands you applied to the coin (stored in the Log). With purses, a problem is that some operations are not reproducible when a coin is isolated - e.g. if you normalise "globally" on a purse using min-max, the coin needs to see the the global min and max (i.e. for all time points). If you just pass the coin on its own, it can't see the other time points, so it cannot reproduce the results.

In short, for the function to work, you need to be able to run `Regen()` on the coin successfully. It might also depend if you have included some custom operations and they cannot be reproduced.
